### PR TITLE
Python3 compatibility

### PR DESCRIPTION
**Pull request by petricm**
BEGINRELEASENOTES
- Make python code compatible to python 2 and 3
  - add `absolute_import` and `unicode_literals` to all files
    - fix API calls and cast `unicode` to `string` when needed
  - replace print statement with logging
  - remove old octal literal
  - use future division
  - use `six`:
     - replace `dict.iteritems` with `six.iteritems`
     - replace `xrange` with `range` from `six.move`
     - replace `basestring` with `six.string_types`
     - replace `raw_input` with `input` from `six.moves`
     - added a copy of six.py named ddsix.py to DDCore
  - Replace deprecated `execfile` with call to `open`, `compile` and `exec`
  - Remove usage of `apply`
  - use `io.open` instead of standard `open`
  - convert `except a,b` to `except a as b`
  - change `dict.has_key` to `key in dict`
- Require DD4hep Python3 CI tests to pass
- Remove obsolete `lcdd.py`
- Remove deprecated `SystemOfUnits.py` and replace everywhere with `g4units.py`
- Flake8 all files

ENDRELEASENOTES

 - Commits in this PR untill https://github.com/AIDASoft/DD4hep/pull/540/commits/6cf772301792ef1d63aff8ee3b46f3d5f8162143 are for `pylint --py3k` issue fixing.

### CODE REVIEW

**Comment by petricm** [comment]
For further development the python 3 compatibility the DD4hep cmake update of @andresailer is needed

**Comment by petricm** [comment]
I am putting this into review. The python detection is now done like https://github.com/AIDASoft/DD4hep/pull/540/commits/f225a01955d94f1679178174b78e9e51005254e1, which prefers Python3 over 2 if both are present. This needs to be addressed once again after [this issue](https://sft.its.cern.ch/jira/browse/ROOT-10293) in ROOT is fixed and we can infer the version from ROOT.

File: DDG4/python/DDSim/bin/ddsim [hunk 1]
```diff
@@ -17,6 +24,4 @@ if __name__ == "__main__":
   except NameError as e:
     if "global name" in str(e):
       globalToSet = str(e).split("'")[1]
-      print """ERROR: Unknown global variable, please add
-      global %s
-      to your steeringFile""" % globalToSet
+      logger.fatal("Unknown global variable, please add global %s to your steeringFile" % globalToSet)
```
Comment (andresailer) [inline DDG4/python/DDSim/bin/ddsim]:
```suggestion
      logger.fatal("Unknown global variable, please add\nglobal %s\nto your steeringFile" % globalToSet)
```

File: DDCore/python/dd4hepFactories.py [hunk 2]
```diff
@@ -10,160 +10,171 @@
 #
 #==========================================================================
 
-import os, sys, optparse, logging
+from __future__ import absolute_import, unicode_literals
+import os
+import sys
+import optparse
+import logging
+from six.moves import input
```
[the remark this answers is not in the document]
Comment (petricm) [reply DDCore/python/dd4hepFactories.py]: Resolved by shipping six with dd4hep as ddsix

File: DDG4/CMakeLists.txt [hunk 3]
```diff
@@ -50,23 +50,23 @@ dd4hep_add_plugin(DDG4Plugins
 #---------------------------  Plugin library for the simulation framework  ---------
 
 
-find_package(Python2 COMPONENTS Development)
-if(TARGET Python2::Python)
+find_package(Python COMPONENTS Development)
```
[the remark this answers is not in the document]
Comment (petricm) [reply DDG4/CMakeLists.txt]: you can always overwrite with the env variable

**Pull request merged**